### ISSUE TITLE
Support for controlled-vocabulary - Hierarchical facets

### ISSUE THREAD

**Issue by heathergreerklein** [DSpace/dspace-angular#815]
From the release plan spreadsheet; no additional notes were included.
Estimate from release plan: TBD
Expressing interest: none

Depends on REST ticket https://github.com/DSpace/DSpace/issues/2870

**Comment by abollini** [comment]
@tdonohue we are ready to start working on that, it is not a trivial task and estimation can be provided only after some extra analysis. Our best guess at the moment is in the range of 64-80 hrs.
We will propose wireframe to present the solution

**Comment by tdonohue** [comment]
@abollini : That sounds reasonable and I agree that this feature likely will require some wireframes or mockups to discuss it.  I'll add a tentative estimate of 72 hours to this.

**Comment by artlowel** [comment]
As discussed during the last meeting, some screenshots of the way we implemented hierarchical facets.

At the bottom of hierarchical facets, a browse tree link is added: 

<img width="268" alt="Screen Shot 2022-10-27 at 11 33 05" src="https://user-images.githubusercontent.com/1567693/198249227-0ade4403-ada4-42e0-8526-4703c07755b6.png">

Clicking it opens a modal containing [the VocabularyTreeviewComponent](https://github.com/DSpace/dspace-angular/blob/6784ac39f64421ec8ac6cbe3ce1d11d2fc245595/src/app/shared/vocabulary-treeview/vocabulary-treeview.component.ts) from the submission:

<img width="819" alt="Screen Shot 2022-10-27 at 11 33 29" src="https://user-images.githubusercontent.com/1567693/198249883-42427ba1-a8ad-4238-8aa4-ada0ebbc7e6d.png">

Users can browse the tree, and click a node to add it, or type in a search query to show only matching nodes:

<img width="815" alt="Screen Shot 2022-10-27 at 11 33 46" src="https://user-images.githubusercontent.com/1567693/198250826-8cd2774a-0181-4068-8cb9-0c1bedfa779a.png">

Selecting a node adds it as a search filter:

<img width="271" alt="Screen Shot 2022-10-27 at 11 34 09" src="https://user-images.githubusercontent.com/1567693/198250969-246f8dae-7b02-43b9-aab2-3f9a1290a1c5.png">

**Comment by tdonohue** [comment]
As discussed in today's DevMtg, we are reassigning this to Atmire for the 7.5 release to implement the [simple solution presented in the screenshots above](https://github.com/DSpace/dspace-angular/issues/815#issuecomment-1293270195).  The rough estimate was that would take only around 20hrs for both frontend & backend work (please correct me if that estimate is incorrect, @artlowel )

A new ticket will be created for future redesign work (as proposed by 4Science) as we all agreed that this initial solution is only a first step.  We agreed that more detailed discussion is needed to enhance this in the future, along the lines of what @abollini  [already noted above](https://github.com/DSpace/dspace-angular/issues/815#issuecomment-1284196054)... that will likely take much more work/effort (rough estimate above was up to 80hrs).
